gapic-generator/lib/gapic/schema/wrappers.rb

```diff
@@ -780,11 +783,11 @@ class Resource
       def initialize descriptor
         @parent = nil
         @descriptor = descriptor
-        @parsed_patterns = descriptor.pattern.map do |pattern|
-          pattern.split("/").map do |segment|
-            segment =~ %r{\{[^/\}]+(=[^\}]+)?\}} ? "*" : segment
-          end.freeze
+        patterns = descriptor.pattern.map do |pattern|
+          Gapic::PathPattern.parse pattern
         end.freeze
+        @parsed_patterns = patterns.map(&:template).freeze
+        @parsed_parent_patterns = patterns.map(&:parent_template).freeze
```

Comment:
Please remove empty parent patterns and duplicates. The previous logic did so, and downstream algorithms might be depending on this.